contracts/integrations/test/forwarder/forwarder_test.ts

```diff
@@ -605,32 +649,39 @@ blockchainTests('Forwarder integration tests', env => {
                 takerAssetAmount: toBaseUnitAmount(11),
             });
             await testFactory.marketBuyTestAsync([order], 0.33, {
-                forwarderFeePercentage: new BigNumber(2),
+                forwarderFeeAmounts: [toBaseUnitAmount(0.2)],
+                forwarderFeeRecipientAddresses: [randomAddress()],
             });
         });
-        it('should fail if the fee is set too high', async () => {
+        it('should fail if there is not enough ETH remaining to complete the fill', async () => {
             const order = await maker.signOrderAsync();
-            const revertError = new ExchangeForwarderRevertErrors.FeePercentageTooLargeError(
-                getPercentageOfValue(constants.PERCENTAGE_DENOMINATOR, new BigNumber(6)),
+            const forwarderFeeAmounts = [toBaseUnitAmount(1)];
+            const revertError = new ExchangeForwarderRevertErrors.CompleteBuyFailedError(
+                order.takerAssetAmount,
+                constants.ZERO_AMOUNT,
             );
             await testFactory.marketBuyTestAsync([order], 0.5, {
-                forwarderFeePercentage: new BigNumber(6),
+                ethValueAdjustment: -1,
+                forwarderFeeAmounts,
+                forwarderFeeRecipientAddresses: [randomAddress()],
                 revertError,
             });
         });
         it('should fail if there is not enough ETH remaining to pay the fee', async () => {
             const order = await maker.signOrderAsync();
-            const forwarderFeePercentage = new BigNumber(2);
-            const ethFee = getPercentageOfValue(
-                order.takerAssetAmount.times(0.5).plus(DeploymentManager.protocolFee),
-                forwarderFeePercentage,
+            const forwarderFeeAmounts = [toBaseUnitAmount(1)];
+            const value = forwarderFeeAmounts[0].minus(1);
+            const revertError = new ExchangeForwarderRevertErrors.InsufficientEthForFeeError(
+                forwarderFeeAmounts[0],
+                value,
             );
-            const revertError = new ExchangeForwarderRevertErrors.InsufficientEthForFeeError(ethFee, ethFee.minus(1));
-            await testFactory.marketBuyTestAsync([order], 0.5, {
-                ethValueAdjustment: -1,
-                forwarderFeePercentage,
-                revertError,
-            });
+            return expect(
+                forwarder
+                    .marketBuyOrdersWithEth([order], order.makerAssetAmount, [order.signature], forwarderFeeAmounts, [
+                        randomAddress(),
+                    ])
+                    .awaitTransactionSuccessAsync({ value }),
+            ).to.revertWith(revertError);
```

Comment:
can we pass the revert error to `marketBuyTestAsync` here? With `fractionalNumberOfOrdersToFill=0` and `ethValueAdjustment: -1`